app/questionnaire/router.py

```diff
@@ -89,6 +96,28 @@ def get_first_incomplete_location_in_survey(self):
         last_block_id = self._schema.get_last_block_id_for_section(all_section_ids[-1])
         return Location(block_id=last_block_id)
 
+    def get_first_incomplete_location_for_section(self, section_id, routing_path):
+        if section_id in self._progress_store.sections:
+            for location in routing_path:
+                if (
+                    location
+                    not in self._progress_store.sections[section_id]['locations']
+                ):
+                    return location
+        return routing_path[0]
+
+    def get_last_complete_location(self, section_id, routing_path):
+        if section_id in self._progress_store.sections:
+            for location in routing_path[::-1]:
+                if location in self._progress_store.sections[section_id]['locations']:
+                    return location
+
+    def get_survey_status(self):
+        if self.is_survey_complete():
+            return CompletionStatus.COMPLETED
+
+        return CompletionStatus.IN_PROGRESS
```

Comment:
I don't see the value in this being a status rather than a boolean. A survey is either complete or it's not.